service/controller/app/resource/status/create.go

```diff
@@ -37,26 +37,6 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return nil
 	}
 
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("finding status for chart %#q in namespace %#q", cr.Name, r.chartNamespace))
-
-	chart, err := cc.Clients.K8s.G8sClient().ApplicationV1alpha1().Charts(r.chartNamespace).Get(ctx, cr.Name, metav1.GetOptions{})
-	if apierrors.IsNotFound(err) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("did not find chart %#q in namespace %#q", cr.Name, r.chartNamespace))
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-		return nil
-	} else if tenant.IsAPINotAvailable(err) {
-		// We should not hammer tenant API if it is not available, the tenant cluster
-		// might be initializing. We will retry on next reconciliation loop.
-		r.logger.LogCtx(ctx, "level", "debug", "message", "tenant cluster is not available.")
-		r.logger.LogCtx(ctx, "level", "debug", "message", "canceling resource")
-		return nil
-	} else if err != nil {
-		return microerror.Mask(err)
-	}
-
-	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("found status for chart %#q in namespace %#q", cr.Name, r.chartNamespace))
-
-	chartStatus := key.ChartStatus(*chart)
 	var desiredStatus v1alpha1.AppStatus
 	if cc.Status.ChartStatus.Status != "" {
```

Comment:
Check `context` first would prevent to find chart CR in tenant clusters.

Reply:
Minor but please add a newline before the if for readability.